### DISCUSSION TITLE
Conda envs not triggering

### DISCUSSION THREAD

**Discussion by Ulthran** [renovatebot/renovate#20556] [Request Help]
BEGIN QUOTE

Thanks for the extra information and the example config. With this I managed to get a PR triggered for an update of a conda environment.
From the logs:
```
DEBUG: Matched 33 file(s) for manager regex: src/envs/amos.yaml, src/envs/antismash.yaml, src/envs/bamm.yaml, src/envs/bbmap.yaml, src/envs/bedtools.yaml, src/envs/bigscape.yaml, src/envs/bigslice.yaml, src/envs/bwa.yaml, src/envs/cat.yaml, src/envs/checkm.yaml, src/envs/concoct.yaml, src/envs/coverm.yaml, src/envs/dastool.yaml, src/envs/fraggenescan.yaml, src/envs/groopm.yaml, src/envs/khmer.yaml, src/envs/kraken.yaml, src/envs/mash.yaml, src/envs/megahit.yaml, src/envs/metabat.yaml, src/envs/minimap2.yaml, src/envs/mmgenome.yaml, src/envs/mmgenome_prepare.yaml, src/envs/pigz.yaml, src/envs/prodigal.yaml, src/envs/quast.yaml, src/envs/report.yaml, src/envs/samtools.yaml, src/envs/seqtk.yaml, src/envs/spades.yaml, src/envs/tree.yaml, src/envs/vamb.yaml, src/envs/vsearch.yaml

DEBUG: 2 flattened updates found: bioconda/spades, bioconda/vsearch
DEBUG: Returning 2 branch(es)
DEBUG: Fetching changelog: https://github.com/ablab/spades (3.14.0 -> 3.15.4)
```

And 2 PRs for 2 packages that I enabled: https://github.com/mdehollander/orochi/pulls

![image](https://user-images.githubusercontent.com/262872/156830355-867da2d7-66b9-4788-bf8a-4b65388d48dd.png)

To get the environment files in subfolders recognized I changed the regular expression in the config to:
```
      "fileMatch": [
        "^(?:src/envs/)?\\w+\\.yaml$"
      ],
```
Looks very promising! Thanks! Looking forward for a manager :)

_Originally posted by @mdehollander in https://github.com/renovatebot/renovate/issues/2213#issuecomment-1059460247_

END QUOTE

I tried to follow this issue to auto update conda envs. My `renovate.json` looks like this:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
      "automerge": true
    }
  ],
  "labels": [
    "renovate",
    "dependencies"
  ],
  "regexManagers": [
    {
      "description": "Upgrade conda dependencies",
      "fileMatch": [
        "^(?:workflow/envs/)?\\w+\\.yaml$"
      ],
      "matchStrings": [
        "# renovate datasource=conda\\sdepName=(?<depName>.*?)\\s+- [a-z0-9]+==\"?(?<currentValue>.*)\"?"
      ],
      "datasourceTemplate": "conda"
    }
  ]
}
```

My conda env files are all in `workflow/envs/` and have names like `iqtree.yaml`. Here's what that env file looks like:

```
channels:
  - bioconda
  - conda-forge
dependencies:
  - conda-forge::libgcc-ng>=9.4.0
  # renovate datasource=conda depName=bioconda/iqtree
  - bioconda::iqtree=2.2.0_beta
```

Any thoughts on why only the renovate defaults are triggering?

EDIT

This line shows up in the logs: `DEBUG: Matched 4 file(s) for manager regex: workflow/envs/iqtree.yaml, workflow/envs/muscle.yaml, workflow/envs/openjdk.yaml, workflow/envs/raxml.yaml` but then there's no further mention of the regex manager or any of those files.

**Comment by rarkins** [comment]
Please don't create issues in future
